jablib/src/main/java/org/jabref/model/groups/DirectoryGroup.java

```diff
@@ -0,0 +1,238 @@
+                if (childNode.getGroup() instanceof DirectoryGroup childGroup) {
+                    allPDFs.addAll(childGroup.getAllPDFs());
+                }
+            }
+        }
+        return allPDFs;
+    }
+
+    public Optional<GroupTreeNode> getNode() {
+        Optional<GroupTreeNode> groupNode = Optional.empty();
+        Optional<GroupTreeNode> rootNode = database.getMetaData().getGroups();
+        if (rootNode.isPresent()) {
+            List<GroupTreeNode> groupNodeCandidates = rootNode.get().findChildrenSatisfying(groupTreeNode -> groupTreeNode.getGroup().equals(this));
+            if (groupNodeCandidates.size() == 1) {
+                groupNode = Optional.of(groupNodeCandidates.getFirst());
+            }
+        }
+        return groupNode;
+    }
+
+    public Boolean isDeleted() {
+        return getNode().isEmpty();
+    }
+
+    @Override
+    public boolean contains(BibEntry entry) {
+        List<LinkedFile> entryFiles = entry.getFiles();
+        for (LinkedFile linkedFile : entryFiles) {
+            Path parentPath = Path.of(linkedFile.getLink()).getParent();
+            if (parentPath != null && absoluteDirectoryPath.toString().endsWith(parentPath.toString())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public boolean isDynamic() {
+        return true;
+    }
+
+    @Override
+    public AbstractGroup deepCopy() {
+        return new DirectoryGroup(name.getValue(), context, absoluteDirectoryPath, directoryUpdateMonitor, database, user);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if ((o == null) || (getClass() != o.getClass())) {
+            return false;
+        }
+        if (!super.equals(o)) {
+            return false;
+        }
+        DirectoryGroup group = (DirectoryGroup) o;
+        return Objects.equals(absoluteDirectoryPath, group.absoluteDirectoryPath);
+    }
+
+    @Override
+    public String toString() {
+        return "DirectoryGroup{" +
+                "directoryPath=" + absoluteDirectoryPath +
+                ", directoryUpdateMonitor=" + directoryUpdateMonitor +
+                "} " + super.toString();
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(super.hashCode(), absoluteDirectoryPath);
+    }
+
+    public Path getAbsoluteDirectoryPath() {
+        return absoluteDirectoryPath;
+    }
+
+    public BibDatabaseContext getBibDatabaseContext() {
+        return this.database;
+    }
+
+    public void setBibDatabaseContext(BibDatabaseContext bibDatabaseContext) {
+        this.database = bibDatabaseContext;
+    }
+
+    private void cleanupFilesOfAnEntry(BibEntry entry) {
+        List<LinkedFile> cleanedFiles = new ArrayList<>();
+        for (LinkedFile file : entry.getFiles()) {
+            if (Files.exists(Path.of(file.getLink()).toAbsolutePath())) {
```

Comment:
This might work for DirectoryGroups as imlemented for now - but it won't work if relative paths are used.